src/go/rpk/pkg/kafka/utils.go

```diff
@@ -0,0 +1,256 @@
+package kafka
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/burdiyan/kafkautil"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/cmd/container/common"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"golang.org/x/sync/errgroup"
+)
+
+// internal topics
+const CoprocessorTopic = "coprocessor_internal_topic"
+
+func ContainerBrokers() []string {
+	c, err := common.NewDockerClient()
+	if err != nil {
+		log.Debug(err)
+		return []string{}
+	}
+	nodes, err := common.GetExistingNodes(c)
+	if err != nil {
+		log.Debug(err)
+		return []string{}
+	}
+	grp := errgroup.Group{}
+	mu := sync.Mutex{}
+	addrs := []string{}
+	for _, node := range nodes {
+		s := node
+		grp.Go(func() error {
+			mu.Lock()
+			defer mu.Unlock()
+			addrs = append(
+				addrs,
+				fmt.Sprintf(
+					"127.0.0.1:%d",
+					s.HostKafkaPort,
+				),
+			)
+			return nil
+		})
+	}
+	err = grp.Wait()
+	if err != nil {
+		log.Debug(err)
+		return []string{}
+	}
+	return addrs
+}
+
+func DeduceBrokers(
+	fs afero.Fs, configuration func() (*config.Config, error), brokers *[]string,
+) func() []string {
+	return func() []string {
+		bs := *brokers
+		// Prioritize brokers passed through --brokers
+		if len(bs) != 0 {
+			log.Debugf("Using --brokers: %s", strings.Join(bs, ", "))
+			return bs
+		}
+		// Otherwise, try to detect if a local container cluster is
+		// running, and use its brokers' addresses.
+		bs = ContainerBrokers()
+		if len(bs) > 0 {
+			log.Debugf(
+				"Using container cluster brokers %s",
+				strings.Join(bs, ", "),
+			)
+			return bs
+		}
+		// Otherwise, try to find an existing config file.
+		conf, err := configuration()
+		if err != nil {
+			log.Trace(
+				"Couldn't read the config file." +
+					" Assuming 127.0.0.1:9092",
+			)
+			log.Debug(err)
+			return []string{"127.0.0.1:9092"}
+		}
+
+		// Add the seed servers' Kafka addrs.
+		if len(conf.Redpanda.SeedServers) > 0 {
+			for _, b := range conf.Redpanda.SeedServers {
+				addr := fmt.Sprintf(
+					"%s:%d",
+					b.Host.Address,
+					conf.Redpanda.KafkaApi.Port,
+				)
+				bs = append(bs, addr)
+			}
+		}
+		// Add the current node's Kafka addr.
+		selfAddr := fmt.Sprintf(
+			"%s:%d",
+			conf.Redpanda.KafkaApi.Address,
+			conf.Redpanda.KafkaApi.Port,
+		)
+		bs = append(bs, selfAddr)
+		log.Debugf(
+			"Using brokers from config: %s",
+			strings.Join(bs, ", "),
+		)
+		return bs
+	}
+}
+
+func CreateProducer(
```

Comment:
All these closure-returning funcs are for the CLI layer. I don't think they belong here. Same for `CreateClient`, `CreateAdmin`

Reply:
Yes, but those functions create the Client and Admin for kafka, I think is a good place, but if not, where should it be? `common`

Reply:
Yup, they should go there too.

Reply:
Done